### PR TITLE
6702 solaris compilation

### LINKED ISSUE #6702: Compilation link for Solaris/hpux fail
# Description
This issue aims to solve the compilation link for these systems, apparently the drawback here is that solaris and hpux are being considered as a linux system, when it is a unix based system independent of linux or bsd systems.

image

#Dod
- [x] Change CMakelists to compile Unixbased systems providers
- [x] Change code (add unix based provider only could return osinfo at this moment)
- [x] Run RTR
- [x] PR

### PR DESCRIPTION
**Pull request by Dwordcito**
|Related issue|
|---|
|Closes #6702 |

Fix solaris compilation and runtime exception.

RTR results
image
image

### CODE REVIEW

File: src/wazuh_modules/syscollector/src/syscollectorImp.cpp [hunk 1]
```diff
@@ -543,75 +543,78 @@ void Syscollector::scanNetwork()
...
-                    protoTableData["dhcp"]    = addressTableData.at("dhcp");
-                    protoTableData["metric"]  = addressTableData.at("metric");
-
-                    // "network_address" table data to update and notify
-                    addressTableData["iface"]   = item.at("name");
-                    addressTableData["proto"]   = "IPv4";
-                    addressTableDataList.push_back(addressTableData);
-                }
+                    // Split the resulting networks data into the specific DB tables
+                    // "network_iface" table data to update and notify
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/src/syscollectorImp.cpp:563]: This comment should be updated as: **dbsync_** network_iface

File: src/wazuh_modules/syscollector/src/syscollectorImp.cpp [hunk 2]
```diff
@@ -543,75 +543,78 @@ void Syscollector::scanNetwork()
...
+                    ifaceTableData["mac"]        = item.at("mac");
+                    ifaceTableData["tx_packets"] = item.at("tx_packets");
+                    ifaceTableData["rx_packets"] = item.at("rx_packets");
+                    ifaceTableData["tx_errors"]  = item.at("tx_errors");
+                    ifaceTableData["rx_errors"]  = item.at("rx_errors");
+                    ifaceTableData["tx_dropped"] = item.at("tx_dropped");
+                    ifaceTableData["rx_dropped"] = item.at("rx_dropped");
+                    ifaceTableDataList.push_back(ifaceTableData);
+
+                    // "network_protocol" table data to update and notify
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/src/syscollectorImp.cpp:579]: This comment should be updated as: **dbsync_**network_protocol

File: src/wazuh_modules/syscollector/src/syscollectorImp.cpp [hunk 3]
```diff
@@ -543,75 +543,78 @@ void Syscollector::scanNetwork()
...
+                    protoTableData["type"]    = item.at("type");
+                    protoTableData["gateway"] = item.at("gateway");
+
+                    if (item.find("IPv4") != item.end())
+                    {
+                        nlohmann::json addressTableData(item.at("IPv4"));
+                        protoTableData["dhcp"]    = addressTableData.at("dhcp");
+                        protoTableData["metric"]  = addressTableData.at("metric");
+
+                        // "network_address" table data to update and notify
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/src/syscollectorImp.cpp:591]: This comment should be updated as: **dbsync_**network_address

File: src/wazuh_modules/syscollector/src/syscollectorImp.cpp [hunk 4]
```diff
@@ -543,75 +543,78 @@ void Syscollector::scanNetwork()
...
-                    addressTableData["iface"] = item.at("name");
-                    addressTableData["proto"] = "IPv6";
-                    addressTableDataList.push_back(addressTableData);
+                    if (item.find("IPv6") != item.end())
+                    {
+                        nlohmann::json addressTableData(item.at("IPv6"));
+                        protoTableData["dhcp"]    = addressTableData.at("dhcp");
+                        protoTableData["metric"]  = addressTableData.at("metric");
+
+                        // "network_address" table data to update and notify
```
Comment (tinachoB) [inline src/wazuh_modules/syscollector/src/syscollectorImp.cpp:603]: This comment should be updated as: **dbsync_**network_address

**Review by tinachoB — CHANGES_REQUESTED** [review]

**Review by tinachoB — APPROVED** [review]
LGTM!

**Review by konangc87 — APPROVED** [review]

**Pull request merged**